### DISCUSSION TITLE
Example for Cut Stock Problem

### DISCUSSION THREAD

**Discussion by zahidraza** [TimefoldAI/timefold-solver#813] [Ideas]
Currently we have quickstart/example for each use cases mentioned in documentation introduction except Cutting Stock Problem. Cutting Stock Problem is completely different from other use cases as it involves 2D optimisation. I have an use case which falls under Cutting Stock problem. My use case is:

**Input:** 
- Fabric Roll: Fabric in Roll form where width of fabric is fixed but length is not fixed. 
- Marker:  Marker is group of garment of different sizes. e.g :- Marker of 4 garments in size - S, M, L, XL. Each garment will have multiple part of different shape like Front, Back, Pocket etc.
- There can be some constraints w.r.t shape arrangement as per business use case

**Output:**
We need to arrange all the shapes of Marker in such a way that all parts are arranged in minimum possible Fabric Length. In other words minimising waste due to gap between shapes

Since, This use case involves 2D optimisation and current all examples are 1D optimisation, I am not getting clue on how to design the Domain model for my use case. If there was any example related to Cutting Stock Problem, it would have given some direction to design domain model.

**Comment by ge0ffrey** [comment]
Optimizing 2D surface (or 3D volume) packing well requires running a Local Search that orders a list which is used as input for First Fit Decreasing.

Timefold Solver doesn't support this out-of-the-box yet. There's no quickstarts for it. It's also not on our short term roadmap.

That being said, I did prove it's possible once, back when Timefold Solver was still called OptaPlanner, by hacking it in for 3D volume packing: https://github.com/ge0ffrey/packing-santas-sleigh
Here be dragons.
